src/StreamState.h

```diff
@@ -21,15 +21,13 @@ using StreamId = uint32_t;
 
 class StreamState {
  public:
-  StreamState()
-      : resumeTracker_(new ResumeTracker()), resumeCache_(new ResumeCache()) {}
-
+  StreamState() = default;
```

Comment:
Please remove the ctor and dtor completely. The compiler will generate them automatically.